app/scripts/controllers/pod.js

```diff
@@ -191,6 +193,10 @@ angular.module('openshiftConsole')
       return terminals;
     };
 
+    var updateContainersYet = function(pod) {
+      $scope.noContainersYet = $scope.containersRunning(pod.status.containerStatuses) === 0; 
```

Comment:
hmm as it is now i think this is going cause the terminals to disappear if you had already connected terminals that were deleted under the covers.  think in the suggested updates it might have messed up your previous logic.  can you re-test that scenario?